### PR TITLE
Implemented Get Messsage feature in wchat

### PR DESCRIPTION
**Pull request by Rits1272**
## Issue that this pull request solves

Closes: #77 

## Types of changes

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

## Screenshots

Screenshot (61)

## Other information

A user can now receive messages in `wchat` . 
The script will be continuously listening to messages from the target. If a message is sent by the target it will be displayed in the terminal. If a user wants to send a message he simply needs to press `space bar`. After that, the user can send his message.

Initially, @xandao6 suggested using threading. But since `asyncio` is itself always single-threaded it was no feasible to add a thread. So I came out with this solution.

### CODE REVIEW

**Comment by Rits1272** [comment]
Please review @xandao6 @rpotter12 @priyansh19. Sorry for sending PR late. It actually took me a while to come out with the solution.

File: wplay/terminal_chat.py [hunk 1]
```diff
@@ -31,22 +33,56 @@
...
+                    else:
+                        await target_select.manual_select_target(page)
+                    message = io.ask_user_for_message_breakline_mode()       
+
+                #File Share:
+                if '#_FILE' in message:
+                    message.remove('#_FILE')
+                    await io.send_file(page)
+                await io.send_message(page, message)
+        except:
```
Comment (rpotter12) [inline wplay/terminal_chat.py:60]: Codacy Issue found: [Try, Except, Pass detected.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5335824)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- wplay/terminal_chat.py  3
         

Clones removed
==============
+ wplay/terminal_chat.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5335824&bid=17466134)

File: wplay/terminal_chat.py [hunk 2]
```diff
@@ -31,22 +33,56 @@
...
+        last_message_time = last_message_time[0].replace('[', '')
+        lastMessage = lastMessage.replace(last_message_time, '')
+    except Exception as e:
+        print(e)
+        lastMessage = ""
+    global lastOutgoingMessage
+    if tg.lower() in last_message_sender.lower() and lastOutgoingMessage!=lastMessage:
+        print(Fore.GREEN + f"{tg}-", end="")
+        print(lastMessage, end="")
+        print(Style.RESET_ALL)
```
Comment (rpotter12) [inline wplay/terminal_chat.py:87]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5335824)

File: wplay/terminal_chat.py [hunk 3]
```diff
@@ -31,22 +33,56 @@
...
+                message = io.ask_user_for_message_breakline_mode()
+                if '...' in message:
+                    message.remove('...')
+                    await io.send_message(page, message)
+                    target = input("\n\nNew Target Name: ")
+                    if target is not None:
+                        await target_search.search_and_select_target(page, target)
+                    else:
+                        await target_select.manual_select_target(page)
+                    message = io.ask_user_for_message_breakline_mode()
```
Comment (rpotter12) [inline wplay/terminal_chat.py:53]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5335824)

File: wplay/terminal_chat.py (hunk 1 above)
Comment (rpotter12) [inline wplay/terminal_chat.py:60]: Codacy Issue found: [No exception type(s) specified](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5335824)

**Comment by Rits1272** [comment]
Hello @rpotter12 I have mentioned in the screenshot as well as in the description if you want to send message you first need to hit `spacebar`. Kindly check.

**Comment by xandao6** [comment]
add keyboard library to setup.py also.

**Comment by xandao6** [comment]
image

**Comment by xandao6** [comment]
I got a lot of bugs using 'q' to exit,

**Comment by rpotter12** [comment]
> Hello @rpotter12 I have mentioned in the screenshot as well as in the description if you want to send message you first need to hit `spacebar`. Kindly check :)

@Ritz1272 I am still not able to send message from your implementation. Please check your implementation again and test it again.

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- wplay/terminal_chat.py  2
         

Clones removed
==============
+ wplay/terminal_chat.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5335824&bid=17466134)

File: wplay/terminal_chat.py [hunk 4]
```diff
@@ -27,26 +29,58 @@ async def chat(target):
...
+        last_message_time = last_message_time[0].replace('[', '')
+        lastMessage = lastMessage.replace(last_message_time, '')
+    except Exception as e:
+        print(e)
+        lastMessage = ""
+    global lastOutgoingMessage
+    if tg.lower() in last_message_sender.lower() and lastOutgoingMessage!=lastMessage:
+        print(Fore.GREEN + f"{tg}-", end="")
+        print(lastMessage, end="")
+        print(Style.RESET_ALL)
```
Comment (rpotter12) [inline wplay/terminal_chat.py:85]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5335824)

File: wplay/terminal_chat.py [hunk 5]
```diff
@@ -27,26 +29,58 @@ async def chat(target):
...
+                message = io.ask_user_for_message_breakline_mode()
+                if '...' in message:
+                    message.remove('...')
+                    await io.send_message(page, message)
+                    target = input("\n\nNew Target Name: ")
+                    if target is not None:
+                        await target_search.search_and_select_target(page, target)
+                    else:
+                        await target_select.manual_select_target(page)
+                    message = io.ask_user_for_message_breakline_mode()
```
Comment (rpotter12) [inline wplay/terminal_chat.py:51]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5335824)

File: wplay/terminal_chat.py [hunk 6]
```diff
@@ -27,26 +29,58 @@ async def chat(target):
...
+                    else:
+                        await target_select.manual_select_target(page)
+                    message = io.ask_user_for_message_breakline_mode()       
+
+                #File Share:
+                if '#_FILE' in message:
+                    message.remove('#_FILE')
+                    await io.send_file(page)
+                await io.send_message(page, message)
+        except:
```
Comment (rpotter12) [inline wplay/terminal_chat.py:58]: Codacy Issue found: [No exception type(s) specified](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5335824)

Comment (rpotter12) [inline wplay/terminal_chat.py:58]: Codacy Issue found: [Try, Except, Pass detected.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5335824)

**Comment by Rits1272** [comment]
- Manual target select does not give any errors
- Added `keyboard` library in the `setup.py` file
-  Removed the option to exit the terminal chat by pressing `q`

Please review @rpotter12 @xandao6

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- wplay/terminal_chat.py  2
         

Clones removed
==============
+ wplay/terminal_chat.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5335824&bid=17466134)

File: wplay/terminal_chat.py [hunk 7]
```diff
@@ -25,31 +27,60 @@ async def chat(target):
...
+        last_message_time = last_message_time[0].replace('[', '')
+        lastMessage = lastMessage.replace(last_message_time, '')
+    except Exception as e:
+        print(e)
+        lastMessage = ""
+    global lastOutgoingMessage
+    if tg.lower() in last_message_sender.lower() and lastOutgoingMessage!=lastMessage:
+        print(Fore.GREEN + f"{tg}-", end="")
+        print(lastMessage, end="")
+        print(Style.RESET_ALL)
```
Comment (rpotter12) [inline wplay/terminal_chat.py:85]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5335824)

File: wplay/terminal_chat.py [hunk 8]
```diff
@@ -25,31 +27,60 @@ async def chat(target):
...
+                    else:
+                        await target_select.manual_select_target(page)
+                    message = io.ask_user_for_message_breakline_mode()       
+
+                #File Share:
+                if '#_FILE' in message:
+                    message.remove('#_FILE')
+                    await io.send_file(page)
+                await io.send_message(page, message)
+        except:
```
Comment (rpotter12) [inline wplay/terminal_chat.py:58]: Codacy Issue found: [No exception type(s) specified](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5335824)

File: wplay/terminal_chat.py [hunk 9]
```diff
@@ -25,31 +27,60 @@ async def chat(target):
...
+                message = io.ask_user_for_message_breakline_mode()
+                if '...' in message:
+                    message.remove('...')
+                    await io.send_message(page, message)
+                    target = input("\n\nNew Target Name: ")
+                    if target is not None:
+                        await target_search.search_and_select_target(page, target)
+                    else:
+                        await target_select.manual_select_target(page)
+                    message = io.ask_user_for_message_breakline_mode()
```
Comment (rpotter12) [inline wplay/terminal_chat.py:51]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5335824)

File: wplay/terminal_chat.py (hunk 8 above)
Comment (rpotter12) [inline wplay/terminal_chat.py:58]: Codacy Issue found: [Try, Except, Pass detected.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5335824)

**Comment by rpotter12** [comment]
Screenshot from 2020-04-17 17-15-34

My screenshot doesnot match with your screenshot. `write your message (Enter key for breakline) ('.' alone to send message)` was not found when I ran the software. I think this was your old screenshot.
I am still not able to send message

**Pull request closed without merging**